docs/core-concepts/config.md

```diff
@@ -92,31 +157,186 @@ tools:
+  cd:
+  - type: template              # type template means it's a reference to the pipeline template. read the next section.
+    templateName: cd-pipeline-1
+    options: # optional, used to override defaults in the template
+    vars:    # optional, key/values to be passed to the template
+      key: value
+```
 
-You can override the default value with `-f` or `--config-file`. Examples:
+Since many key/values have default values, it's possible to use the following minimum config for the apps section:
 
-```shell
-dtm apply -f path/to/your/config.yaml
-dtm apply --config-file path/to/your/config.yaml
+```yaml
+apps:
+- name: service-B # minimum config demo
+  spec:
+    language: python
+    framework: django
+  repo:
+    url: github.com/devstream-io/repo-name
+  repoTemplate:
+    url: github.com/devstream-io/repo-name
+  ci:
+    - type: githubactions
+  cd:
+    - type: argocdapp
+```
+
+---
+
+## 6 Template File
+
+You can define some pipeline templates in the template file, like the following:
+
+```yaml
+pipelineTemplates:
+- name: ci-pipeline-1
+  type: githubactions # corresponds to a DevStream plugin
+  options:
+    branch: main      # optional
+    docker:
+      registry:
+        type: dockerhub
+        username: [[ dockerUser ]]
+        repository: [[ app ]]
+- name: cd-pipeline-1
+  type: argocdapp
+  options:
+    app:
+      namespace: argocd
+    destination:
+      server: https://kubernetes.default.svc
+      namespace: default
+    source:
+      valuefile: values.yaml
+      path: helm/[[ app ]]
+      repoURL: ${{repo-scaffolding.myapp.outputs.repoURL}}
+```
+
+Then, you can refer to these pipeline templates in the apps file.
+
+---
+
+## 7 Putting It All Together
+
+Here's a complete example.
+
+Main config file:
+
+```yaml
+toolFile: "./tools.yaml"
+appFile: "./apps.yaml"
+templateFile: "./templates.yaml"
+state:
+  backend: local
+  options:
+    stateFile: devstream.state
+```
+
+Tool file:
+
+```yaml
+tools:
+- name: argocd
+  instanceID: default
 ```
 
-### No Defaults for varFile and toolFile
+App file:
 
-For `varFile` and `toolFile`, no default values are provided.
+```yaml
+apps:
+- name: service-B
+  spec:
+    language: python
+    framework: django
+  repo:
+    url: github.com/devstream-io/repo-name
+  repoTemplate:
+    url: github.com/devstream-io/dtm-scaffolding-python
+  ci:
+    - type: githubactions
+  cd:
+    - type: argocdapp
+```
+
+In this example, we didn't use a var file, or template file, since they are optional and we don't need them in this example.
 
-If `varFile` isn't specified in the main config, `dtm` will not use any var files, even if there is already a file named `variables.yaml` under the current directory.
+## 8 Min Config
 
-Similarly, if `toolFile` isn't specified in the main config, `dtm` will throw an error, even if there is a `tools.yaml` file under the current directory.
+Alternatively, you can merge multiple files into one to achieve a minimum config:
+
+```yaml
+---
```

Comment:
We need to delete this "---" in the document, since the rawConfig is added and all "---" will be removed in the code.

https://github.com/devstream-io/devstream/blob/882a16bc245f1ca8a2c911369882f5d6b276a5e2/internal/pkg/configmanager/rawconfig.go#L48-L51

Reply:
Closed for now after discussion. With or without "---" are all supported now, and we agreed that in the docs we should keep "---".